src/Overlay.js

```diff
@@ -42,56 +35,63 @@ class Overlay extends React.Component {
       , animation: Transition
       , ...props } = this.props;
 
-    let child = null;
-
-    if ( Transition === true ){
+    if (Transition === true) {
       Transition = Fade;
     }
 
-    if (props.show || (Transition && this.state.exiting && !this.state.exited)) {
+    // Don't un-render the overlay while it's transitioning out.
+    const mountOverlay = props.show || (Transition && !this.state.exited);
+
+    if (!mountOverlay) {
+      // Don't bother showing anything if we don't have to.
+      return null;
+    }
 
-      child = children;
+    let child = children;
 
-      // Position the child before the animation to avoid `null` DOM nodes
+    if (Transition) {
+      // This animates the child by injecting props, so it must be inner-most.
       child = (
-        <Position {...{ container, containerPadding, target, placement }}>
-          { child }
-        </Position>
+        <Transition
+          in={props.show}
+          transitionAppear
```

Comment:
why remove the `props.show` here? Originally it was there because you only want the fade in animation to run on mount. if `in=false` initially this will show the overlay then fade it out, instead of just starting closed. Is that the behavior we want?